src/codemagic/cli/argument/common_argument_types.py

```diff
@@ -30,3 +31,20 @@ def json_dict(json_dict: str) -> Dict:
         if not isinstance(d, dict):
             raise argparse.ArgumentTypeError(f'"{json_dict}" is not a dictionary')
         return d
+
+    @staticmethod
+    def iso_8601_date(iso_8601_timestamp: str) -> datetime:
```

Comment:
Static methods defined in `CommonArgumentTypes` are used as argument _type wrappers_ for `argparse` option definitions, and as such I find that it is semantically better to have them named in the way they are now. For example the usage of this method is as:
```python
EARLIEST_RELEASE_DATE = cli.ArgumentProperties(
    key='earliest_release_date',
    flags=('--earliest-release-date',),
    type=cli.CommonArgumentTypes.iso_8601_date,
    ....,
)
```
But I think you are correct that the name should contain `datetime`, not just `date`.

Reply:
Also, the definitions here are not meant to be used outside `argparse` type definitions due to their unique nature of raising `argparse.ArgumentTypeError` in case of invalid input.